File: app/src/test/java/com/digitalisma/boilerplate/util/EventBusTest.java
package com.digitalisma.boilerplate.util;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;

import rx.observers.TestSubscriber;

public class EventBusTest {

    private EventBus eventBus;

    @Rule
    // Must be added to every test class that targets app code that uses RxJava
    public final RxSchedulersOverrideRule overrideSchedulersRule = new RxSchedulersOverrideRule();

    @Before
    public void setUp() {
        eventBus = new EventBus();
    }

    @Test
    public void postedObjectsAreReceived() {
        TestSubscriber<Object> testSubscriber = new TestSubscriber<>();
        eventBus.register(Object.class, testSubscriber);

        Object event1 = new Object();
        Object event2 = new Object();
        eventBus.post(event1);
        eventBus.post(event2);

        testSubscriber.assertValues(event1, event2);
    }

    @Test
    public void onlyReceivesSomeObjects() {
        TestSubscriber<String> testSubscriber = new TestSubscriber<>();
        eventBus.register(String.class, testSubscriber);

        String stringEvent = "Event";
        Integer intEvent = 20;
        eventBus.post(stringEvent);
        eventBus.post(intEvent);

        testSubscriber.assertValueCount(1);
        testSubscriber.assertValue(stringEvent);
    }
}

File: app/src/test/java/eu/se_bastiaan/marietje/util/EventBusTest.kt
package eu.se_bastiaan.marietje.util

import org.junit.Before
import org.junit.Rule
import org.junit.Test

import rx.observers.TestSubscriber

class EventBusTest {

    lateinit var eventBus: EventBus

    @get:Rule
    val overrideSchedulersRule = RxSchedulersOverrideRule()

    @Before
    fun setUp() {
        eventBus = EventBus()
    }

    @Test
    fun postedObjectsAreReceived() {
        val testSubscriber = TestSubscriber<Any>()
        eventBus.register(Any::class.java, testSubscriber)

        val event1 = Any()
        val event2 = Any()
        eventBus.post(event1)
        eventBus.post(event2)

        testSubscriber.assertValues(event1, event2)
    }

    @Test
    fun onlyReceivesSomeObjects() {
        val testSubscriber = TestSubscriber<String>()
        eventBus.register(String::class.java, testSubscriber)

        val stringEvent = "Event"
        val intEvent = 20
        eventBus.post(stringEvent)
        eventBus.post(intEvent)

        testSubscriber.assertValueCount(1)
        testSubscriber.assertValue(stringEvent)
    }
}